### PR TITLE
[vault] unify usage of ureq to clean up stream on request completion

### PR DESCRIPTION
**Pull request by davidiw**
Following up on the previous PR to make this a bit more uniform and deeper understanding of the ureq error codes.

### CODE REVIEW

File: secure/storage/vault/src/lib.rs [hunk 1]
```diff
@@ -130,7 +140,11 @@ impl Client {
                 Ok(policies.policies)
             }
             // There are no policies.
-            404 => Ok(vec![]),
+            404 => {
+                // Explicitly clear buffer so the stream can be re-used.
```
Comment (JoshLind) [inline secure/storage/vault/src/lib.rs:144]: If it were me, I'd define a helper function (e.g., "clear_response_buffer(..)") that contains the comment and the resp.into_string()? call. That way we don't need to duplicate the comment everywhere. 

Of course, the best approach would probably be to wrap ureq somehow to do this automatically for us (that is, assuming we won't get an upstream fix).. You know me and duplication :P

Comment (davidiw) [reply secure/storage/vault/src/lib.rs:144]: Gonna sit on it for now... I would like to see this issue fixed upstream... or to explore creating some sort of droppable wrapper.

Comment (davidiw) [reply secure/storage/vault/src/lib.rs:144]: Thanks for the review!!

**Pull request closed without merging**